packages/webb-ui-components/src/components/ThemeSwitcher/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,117 @@
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+import { Half2Icon, MoonIcon, SunIcon } from '@radix-ui/react-icons';
+import cx from 'classnames';
+import React, { useCallback, useEffect, useState } from 'react';
+
+export interface Props {}
+
+const themes = [
+  {
+    key: 'light',
+    label: 'Light',
+    icon: <SunIcon />,
+  },
+  {
+    key: 'dark',
+    label: 'Dark',
+    icon: <MoonIcon />,
+  },
+];
+
+const ThemeSwitcher = () => {
+  const [preferredTheme, setPreferredTheme] = useState<null | string>(null);
+
+  useEffect(() => {
+    if (
+      localStorage.getItem('theme') === 'dark' ||
+      (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)
+    ) {
+      document.documentElement.classList.add('dark');
+      setPreferredTheme('dark');
+    } else {
+      document.documentElement.classList.remove('dark');
+      setPreferredTheme('light');
+    }
+  }, []);
+
+  const toggleTheme = useCallback((key: string) => {
+    switch (key) {
+      case 'dark': {
+        if (localStorage.getItem('theme') !== key) {
+          document.documentElement.classList.add(key);
+          localStorage.setItem('theme', key);
+        }
+        break;
+      }
+
+      case 'light': {
+        if (localStorage.getItem('theme') !== key) {
+          document.documentElement.classList.remove('dark');
+          localStorage.setItem('theme', key);
+        }
+        break;
+      }
+    }
+  }, []);
```

Comment:
This looks good but can we leverage the react context API, so we can have a single source of truth that is shared across other components